File: pset2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>
#include <ctype.h>

#define logi(x) printf("%d\n", x);


int main(int argc, char *argv[])
{
char *strPlain;
char strCipher[512] = {0};;       // spec says nothing but I think malloc is past curr. class level

	// Only 1 arg allowed
    if (argc != 2)
    {
        printf("Wrong number of arguments. Exiting...\n");
		return 1;
	}
	int keyval = atoi(argv[1]) % 26;
	//logi(keyval)

    // Get plaintext from user
    printf("Plaintext:");
    strPlain = get_string();


    int chr;   //current char in plaintest

    int shift;  // how far to shift from 'A' to 'a'

    // Shift each letter to cipher array
    int len = strlen(strPlain);
    for (int x = 0; x < len; x++)
    {
        chr = strPlain[x];

        if(!isalpha(chr))
        {
            strCipher[x] = chr;
        }
        else
            {
            if (islower(chr))
                shift = 'a';
            else
                shift = 'A';

            strCipher[x] = ((((chr - shift) + keyval) % 26)) + shift;
            }
    }

    printf("ciphertext: %s\n", strCipher);

return (0);
}

File: pset6/caesar/caesar.py
from cs50 import get_string
import sys
from inspect import signature


	# Only 1 arg allowed
if (len(sys.argv) != 2):
    print("Wrong number of arguments. Exiting...");
    sys.exit(1)

iKeyval = int(sys.argv[1]) % 26;
#print("keyval % 26=", iKeyval)

 # Get plaintext from user
print("Plaintext: ", end="");
strPlain = list(get_string())
#print(strPlain)


#ch  #current char in plaintest
#A2aDiff  # how far to shift from 'A' to 'a'
strCipher = ""

#c=(pi+k)mod26

# Shift each letter to cipher array
#len = len(strPlain)
for x in range(0, len(strPlain)):
    ch = strPlain[x];
    #print("x={}->{}".format(x, ch))


    if(not ch.isalpha()):
        strCipher += ch;
    else:
        if (ch.islower()):
            A2aDiff = ord('a')
        else:
            A2aDiff = ord('A')

        ec = chr(((ord(ch) + iKeyval - A2aDiff) %26) + A2aDiff);
        strCipher += ec;

print("ciphertext: ",str(strCipher));